### PR TITLE
Add Sitecore CI post

### PR DESCRIPTION
**Pull request by jflheureux**

### CODE REVIEW

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+    Sitecore.Feature.Accounts.Tests.Attributes.RedirectAuthenticatedAttributeTests.OnActionExecuting_AuthenticatedUser_ShouldRedirect(db: master, item: [], filterContext: ActionExecutingContextProxy { __interceptors = [CastleForwardingInterceptor { ... }], __mixin_NSubstitute_Core_ICallRouter = CallRouter { }, ActionDescriptor = ActionDescriptorProxy { __interceptors = [...], __mixin_NSubstitute_Core_ICallRouter = CallRouter { ... }, ActionName = "", ControllerDescriptor = ControllerDescriptorProxy { ... }, UniqueId = "" }, ActionParameters = [], Controller = null, ... }, redirectAuthenticatedAttribute: RedirectAuthenticatedAttribute { AllowMultiple = False, Order = 185, TypeId = typeof(Sitecore.Feature.Accounts.Attributes.RedirectAuthenticatedAttribute) }) [FAIL]
+      Sitecore.SecurityModel.License.LicenseException : Required license is missing: Runtime
+```
+
+## The Sitecore license
+
+As you don't want to share your Sitecore license to the rest of the world, you don't store it in your repository. The AppVeyor solution for private files is to commit an encrypted version and add a decryption step in the `install` section, before the project is built:
```
Comment (sbelzile-Coveo) [inline _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: and **to** add

step in the `install` => step **to** the `install`

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+  - nuget install secure-file -ExcludeVersion
+  - secure-file\tools\secure-file -decrypt .\lib\license.xml.enc -secret "%LicenseEncryptionKey%"
+```
+
+[secure-file](https://www.appveyor.com/docs/how-to/secure-files/) is an AppVeyor tool to encrypt and decrypt files using a private encryption key. Usage is simple but tricky.
+
+### Encryption key
+
+1. Choose a long enough key for your files to be secure.
+2. Choose a key shorter than 32K characters as it will be stored as an environment variable. There is only 32K to store all the environment variables on Windows. Be considerate.
```
Comment (sbelzile-Coveo) [inline _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: 32K => 32**k** http://lrs.ed.uiuc.edu/students/sbarnett/Science/guidelines.html, sinon tu pourrais écrire le chiffre au complet (32 767)

There is only 32k to store all => There is only 32k **caracters available** to store all (ajoute l'unité de mesure)

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+3. Avoid special characters like ``/ \ | ~ % < > ` ' "`` in your encryption key as it is passed as a command line argument.
```
Comment (sbelzile-Coveo) [inline _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: Le reste de ton énumération est au futur. => it will be passed as

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+4. Avoid the `^` character as AppVeyor will strip it from your environment variable.
+
+### Environment variable
+
+The encryption key must be stored securely in AppVeyor and not be visible in the build logs. Environment variables are perfect for this. It is one of the only setting that is merged between the UI and the configuration file.
+
+![Add an AppVeyor project from GitHub](/images/20170922-appveyor/AppVeyor-EnvironmentVariable.png)
+
+Here I choose to name my environment variable `LicenseEncryptionKey`. You have to set its value.
```
Comment (sbelzile-Coveo) [inline _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: Here**,** I (comma)

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+
+### Installing on your local computer
+
+As you need to encrypt your files locally, you need the tool as well. Run this command in a console:
```
Comment (sbelzile-Coveo) [inline _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: you need the tool as well => you need an encryption tool as well?

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+
+```
+nuget install secure-file -ExcludeVersion
+```
+
+### Encryption
+
+Habitat gulp script copies the Sitecore license file from your webroot to the `lib` folder of the devroot.
+
+Encrypt your Sitecore license file using this command. The location of the file may vary. Replace `YOUR_ENCRYPTION_KEY` by the one you defined earlier.
```
Comment (sbelzile-Coveo) [inline _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: this => the following?

Replace `YOUR_ENCRYPTION_KEY` by the one you defined earlier. => Replace `YOUR_ENCRYPTION_KEY` by **the key** you defined earlier. (On aurait pu écrire: Replace `XXXXXX` by [...])

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+Updating build cache...
+Cache 'packages' - Up to date
+Cache 'C:\Users\appveyor\AppData\Local\NuGet\Cache' - Up to date
+Cache 'C:\Users\appveyor\AppData\Local\NuGet\v3-cache' - Up to date
+Build success
+```
+
+## Improving build performance
+
+By default, the log level is very verbose. For improved build time, you can set the build verbosity in the `build` section:
```
Comment (sbelzile-Coveo) [inline _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: For improved build time => To improve build time

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+
+When a build is successful, AppVeyor can package the artifacts and publish them in its cloud storage. I have yet to try this feature. It will be part of a separate post.
+
+## Conclusion
+
+Before using AppVeyor, I tried to setup another local continuous integration tool. After countless hours, it was still not working. I was discouraged and had hard feelings towards continuous integration.
+
+Needless to say I was happy and impressed by AppVeyor. It took me under 4 hours to learn the basics and have my first successful build of the Sitecore Habitat project. I really like the product as it has a low barrier of entry, and the fact that it is free for public repositories. I plan to use it for a more complex Sitecore project in the near future. Stay tuned as I share my experiences as I progress.
+
+I encourage everybody who believe continuous integration is hard to try AppVeyor and experience how easy it can be.
```
Comment (sbelzile-Coveo) [inline _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: who believe => who believe**s**

J'ai de la difficulté à lire ta phrase. Je pense qu'il manque des virgules. Ex: ma première interprétation a été `continuous integration is hard to try`, puis je me suis rendu compte que ça ne faisait pas de sens.

and experience => and **to** experience

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+tags: [Sitecore, Continuous Integration, AppVeyor, GitHub, NuGet]
+
+author:
+  name: Jean-François L'Heureux
+  bio: Tech Evangelist, Coveo for Sitecore Team
+  twitter: jflh
+  image: jflheureux.jpeg
+---
+
+Continuous integration (CI) is the act of automatically compiling code and running its tests every time a change is made. It is an important step in a project to ensure quality and save time. It needs to be implemented before continuously deploying an application.
```
Comment (sbelzile-Coveo) [inline _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: Other usages of `Continuous integration` => 5x

Other usages of `CI` => 0x

ça ne sert à rien de te définir une abbréviation si tu ne l'utilises pas.

Comment (jflheureux) [reply _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: C'est pour qu'un Google search de CI retourne le post quand même. Comme je ne répète pas souvent "Continuous Integration", j'ai pas utilisé l'abbréviation.

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+
+As you don't want to share your Sitecore license to the rest of the world, you don't store it in your repository. The AppVeyor solution for private files is to commit an encrypted version and add a decryption step in the `install` section, before the project is built:
+
+```yml
+install:
+  - nuget install secure-file -ExcludeVersion
+  - secure-file\tools\secure-file -decrypt .\lib\license.xml.enc -secret "%LicenseEncryptionKey%"
+```
+
+[secure-file](https://www.appveyor.com/docs/how-to/secure-files/) is an AppVeyor tool to encrypt and decrypt files using a private encryption key. Usage is simple but tricky.
```
Comment (amoreauCoveo) [inline _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: Use a capital letter at the beginning of your sentence ;)

Using it is simple, but tricky.

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md (hunk 2 above)
Comment (amoreauCoveo) [reply _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: There **are** only 32k **characters** available

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md (hunk 4 above)
Comment (amoreauCoveo) [reply _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: chose

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md (hunk 5 above)
Comment (amoreauCoveo) [reply _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: I think it's clear that he's referring to secure-file

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 11]
```diff
@@ -0,0 +1,272 @@
...
+
+As you need to encrypt your files locally, you need the tool as well. Run this command in a console:
+
+```
+nuget install secure-file -ExcludeVersion
+```
+
+### Encryption
+
+Habitat gulp script copies the Sitecore license file from your webroot to the `lib` folder of the devroot.
```
Comment (amoreauCoveo) [inline _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: The Habitat gulp script copies . . .

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 12]
```diff
@@ -0,0 +1,272 @@
...
+
+## Packaging build artifacts
+
+When a build is successful, AppVeyor can package the artifacts and publish them in its cloud storage. I have yet to try this feature. It will be part of a separate post.
+
+## Conclusion
+
+Before using AppVeyor, I tried to setup another local continuous integration tool. After countless hours, it was still not working. I was discouraged and had hard feelings towards continuous integration.
+
+Needless to say I was happy and impressed by AppVeyor. It took me under 4 hours to learn the basics and have my first successful build of the Sitecore Habitat project. I really like the product as it has a low barrier of entry, and the fact that it is free for public repositories. I plan to use it for a more complex Sitecore project in the near future. Stay tuned as I share my experiences as I progress.
```
Comment (amoreauCoveo) [inline _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: . . . barrier of entry and is free for public repositories.

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md (hunk 8 above)
Comment (amoreauCoveo) [reply _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: I disagree with sbelzile (except for who believes)
But while I disagree, I see his point. Maybe rephrase as:

If you believe continuous integration is hard, I encourage you to try AppVeyor and experience how easy it can be.

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 13]
```diff
@@ -0,0 +1,272 @@
...
+Discovering tests...OK
+%xunit20%\xunit.console.x86 "C:\projects\habitat\src\Feature\Accounts\Tests\bin\Debug\Ploeh.AutoFixture.Xunit2.dll" -appveyor
+xUnit.net Console Runner (32-bit .NET 4.0.30319.42000)
+System.IO.FileLoadException: Could not load file or assembly 'xunit.core, Version=2.0.0.2929, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
+Command exited with code 1
+```
+
+## Choosing tests
+
+AppVeyor automatically tries to detect the assemblies containing unit tests. In some cases, you must help it in the `test` section:
```
[the remark this answers is not in the document]
Comment (sbelzile-Coveo) [reply _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: prefer => je vais argumenter:

Bien que quand t'as un projet de CI, tes assemblies de tests sont bien définies, AppVeyor, lui, ne le sais pas nécessairement. Il tente de détecter des assemblies contenant des tests en général. C'est pourquoi j'enlèverais le `the`.

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 14]
```diff
@@ -0,0 +1,272 @@
...
+Habitat relies on public NuGet packages from NuGet.org and the Sitecore NuGet feed. These sources are configured in the `nuget.config` file of the Habitat repository and NuGet automatically handles this file. To restore the packages in AppVeyor, you must add a command line instruction to run in the `before_build` section:
+
+```yml
+before_build:
+  - nuget restore Habitat.sln
+```
+
+### Caching
+
+The NuGet packages will rarely change in your project. To speed up the builds, it is recommended to cache the packages between builds. This is done in the `cache` section:
```
[the remark this answers is not in the document]
Comment (sbelzile-Coveo) [reply _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: In your implementation => yes. But, generally speaking, caching packages is a good practice.

File: _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md (hunk 5 above)
Comment (sbelzile-Coveo) [reply _posts/2017-09-22-continuous-integration-of-a-sitecore-project-using-appveyor.md]: 👍 t'as raison.

File: _posts/2017-09-28-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 15]
```diff
@@ -0,0 +1,274 @@
...
+![Add an AppVeyor project from GitHub](/images/20170922-appveyor/AppVeyor-CreateProject.png)
+
+## Configuring the project
+
+AppVeyor has a complete settings user interface, but it alternatively supports storing the project configuration in an `appveyor.yml` file at the root of your repository. I recommend the configuration file for many reasons:
+
+* It provides a complete view of all the configuration, so you do not have to navigate the various UI sections.
+* It is always accessible, even when you work offline.
+* Every time a modification to the file is pushed, your project is automatically built.
+* It allows contributors forking your project to easily use AppVeyor to continuously integrate their work.
```
Comment (francoislg) [inline _posts/2017-09-28-continuous-integration-of-a-sitecore-project-using-appveyor.md]: >  It allows contributors **to fork** your project

Mais vu que ça fais une phrase avec 3 `to`, (`to fork => to easily => to integrate)`, je reformulerais:

> It allows contributors to continuously integrate their work with AppVeyor when they fork your project.

File: _posts/2017-09-28-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 16]
```diff
@@ -0,0 +1,274 @@
...
+
+You should note that the file is not merged with the UI settings except for the environment variables and notification settings. The build version format is taken from the UI if it is not set in the file.
```
Comment (francoislg) [inline _posts/2017-09-28-continuous-integration-of-a-sitecore-project-using-appveyor.md]: > with the **settings UI,** except ...

(flip settings|UI, and +comma)

File: _posts/2017-09-28-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 17]
```diff
@@ -0,0 +1,274 @@
...
+    18 Warning(s)
+    23 Error(s)
+
+Time Elapsed 00:00:11.54
+Command exited with code 1
+```
+
+## Restoring NuGet packages
+
+Habitat relies on public NuGet packages from NuGet.org and from the Sitecore NuGet feed. These sources are configured in the `nuget.config` file of the Habitat repository and NuGet automatically handles this file. To restore the packages in AppVeyor, you must add a command line instruction to run in the `before_build` section:
```
Comment (francoislg) [inline _posts/2017-09-28-continuous-integration-of-a-sitecore-project-using-appveyor.md]: Link for `Sitecore NuGet feed` would be nice :)

https://sitecore.myget.org/gallery/sc-packages

File: _posts/2017-09-28-continuous-integration-of-a-sitecore-project-using-appveyor.md [hunk 18]
```diff
@@ -0,0 +1,274 @@
...
+
+    Sitecore.Feature.Accounts.Tests.Attributes.RedirectAuthenticatedAttributeTests.OnActionExecuting_AuthenticatedUser_ShouldRedirect(db: master, item: [], filterContext: ActionExecutingContextProxy { __interceptors = [CastleForwardingInterceptor { ... }], __mixin_NSubstitute_Core_ICallRouter = CallRouter { }, ActionDescriptor = ActionDescriptorProxy { __interceptors = [...], __mixin_NSubstitute_Core_ICallRouter = CallRouter { ... }, ActionName = "", ControllerDescriptor = ControllerDescriptorProxy { ... }, UniqueId = "" }, ActionParameters = [], Controller = null, ... }, redirectAuthenticatedAttribute: RedirectAuthenticatedAttribute { AllowMultiple = False, Order = 185, TypeId = typeof(Sitecore.Feature.Accounts.Attributes.RedirectAuthenticatedAttribute) }) [FAIL]
+      Sitecore.SecurityModel.License.LicenseException : Required license is missing: Runtime
+```
+
+## The Sitecore license
+
+As you don't want to share your Sitecore license to the rest of the world, you don't store it in your repository. The AppVeyor solution for private files is to commit an encrypted version and to add a decryption step to the `install` section, before the project is built:
```
Comment (francoislg) [inline _posts/2017-09-28-continuous-integration-of-a-sitecore-project-using-appveyor.md]: > **Since** you don't want ..

> you **must** not store it

> **AppVeyor's** solution **to manage** private files

> ... and ~~to~~ add a decryption
